packages/core/src/factory.ts

```diff
@@ -1,65 +1,68 @@
-import { DiagnosticSeverity } from '@stoplight/types';
 import * as Either from 'fp-ts/lib/Either';
 import { getOrElse, map } from 'fp-ts/lib/Option';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TaskEither from 'fp-ts/lib/TaskEither';
 import { defaults } from 'lodash';
-import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, ProblemJsonError } from './types';
+import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, IPrismProxyConfig } from './types';
 import { validateSecurity } from './utils/security';
 
+function isProxyConfig(p: IPrismConfig): p is IPrismProxyConfig {
+  return !p.mock;
```

Comment:
hm.. maybe worth adding?
```suggestion
  return !p.mock && p.proxy;
```

Reply:
proxy is a property that does not exist

Reply:
@XVincentX `upstream` then?

Reply:
We can add it effectively, something like `return !p.mock && 'upstream' in p;` but @StefanDywersant made me remove it :trollface:

Let's keep it this way for now.